roles/translator/src/downstream_sv1/downstream.rs

```diff
@@ -79,80 +86,155 @@ impl Downstream {
         }));
         let self_ = downstream.clone();
 
+        let (tx_shutdown, rx_shutdown): (Sender<bool>, Receiver<bool>) = async_channel::bounded(3);
+
         // Task to read from SV1 Mining Device Client socket via `socket_reader`. Depending on the
```

Comment:
This comment should move down to the loop and you should add a comment about how the new shutdown channels will be used.